lib/api/programData.js

```diff
@@ -23,10 +24,14 @@ export async function FetchProgramData(
 ) {
   switch (req.method) {
     case 'GET':
-      if (!AuthIsDisabled() && !(await AuthIsValid(req)))
+      if (
+        !AuthIsDisabled() &&
+        !(await AuthIsValid(req)) &&
+        !(await getToken({ req }))
```

Comment:
AuthIsValid is using the getSession should that be change to getToken?
Just seems like there's some duplicate validation here. Also maybe only need to getToken once?